kaleidescape/device.py

```diff
@@ -556,6 +560,20 @@ async def _handle_event(self, event: str, *args) -> None:
         elif isinstance(response, messages.CinemascapeMask):
             self._update_cinemascape_mask(response)
 
+        # User-defined events
+        elif isinstance(response, messages.UserDefinedEvent):
+            # User-defined event - dispatch with event type for better monitoring
+            self._dispatcher.send(
+                const.SIGNAL_DEVICE_EVENT, response.device_id, f"{response.name}:{response.event_type}"
+            )
+            return
```

Comment:
Instead of short circuiting here. I would suggest just defining the message to be returned and let the `self._dispatcher.send()` on 577 send it like normal.

```
  elif isinstance(response, messages.UserDefinedEvent):
    dispatch_msg = f"{response.name}:{response.event_type}"

... edit line 577
  self._dispatcher.send(
      const.SIGNAL_DEVICE_EVENT, response.device_id, dispatch_msg
  )
```

And set the default message at line 507
```
response: Response = args[0]
+ dispatch_msg = response.name
```